### ISSUE TITLE
Set JVM Properties in docker-gocd-agent-ubuntu

### ISSUE THREAD

**Issue by rikostave1234** [gocd/gocd#7172]
With the update to docker-gocd-agent-ubuntu-16.04:v19.9 (coming from 19.2), the health checks for our agents started failing. This is due to the fact that system properties passed as environment variable don't get attached to the actual java command. 

We used to pass the following env variable to our docker container:

`
GO_AGENT_SYSTEM_PROPERTIES: '-Dgo.agent.status.api.bind.port=8152 -Dgo.agent.status.api.bind.host=0.0.0.0'`

In the conatiner logs I was able to see the actual java command with those properties attached. After the update to v19.9, the java command doesn't have the properties attached.

In the README for docker-gocd-agent-ubuntu-16.04, the name of the environment variable for passing properties has been changed to GOCD_AGENT_JVM_OPTS. I tried that, but it does not work as well. In docker-entrypoint.sh there is another environment variable with properties (AGENT_BOOTSTRAPPER_JVM_ARGS), same there.

I saw that you started using "Java Service Wrapper" which seems to wrap the start of the JVM. I also tried to add the properties to the wrapper-properties.conf without success (I might have used the wrong prefix for the properties).

Could someone please provide the intended way to pass system properties to the jvm call in v19.9?

Thank you!

**Comment by arvindsv** [comment]
@rikostave1234 I think there's a small bug in the entrypoint script. On [this line](https://github.com/gocd/docker-gocd-agent-ubuntu-16.04/blob/c1f6fd22d7f421564f530ebb8ea3a7f113433da9/docker-entrypoint.sh#L145), it looks like it is not writing that line to the `wrapper-properties.conf` file.

Here's what you should be able to do to work around it for now:

```bash
$ cat <<EOF >set_agent_args.sh
echo 'set.AGENT_STARTUP_ARGS=%AGENT_STARTUP_ARGS% -Dgo.console.stdout=true %GOCD_AGENT_JVM_OPTS%' >>/go-agent/wrapper-config/wrapper.conf
EOF

$ chmod +x set_agent_args.sh

$ docker run -v `pwd`/set_agent_args.sh:/docker-entrypoint.d/set_agent_args.sh -e GOCD_AGENT_JVM_OPTS='-Dgo.agent.status.api.bind.port=8152 -Dgo.agent.status.api.bind.host=0.0.0.0' -e GO_SERVER_URL=https://...
```

@ketan might be able to confirm.

**Comment by rikostave1234** [comment]
Thank you @arvindsv!

Yep, I can confirm that. I just fixed the line in my local compose setup, works fine now!

I did not see that on Friday because I thought that the jvm args for the wrapper have to be passed as `wrapper.java.additional` property...

I will add this to our Dockerfile temporarily. Should I open a PR for this?

**Comment by arvindsv** [comment]
@rikostave1234 Great.

Yes, please. A PR would be nice.

**Comment by arvindsv** [comment]
@rikostave1234 There is some trick to it, though. These Dockerfiles are generated from a template somewhere. Let me find it.

**Comment by arvindsv** [comment]
@rikostave1234 It is here: https://github.com/gocd/gocd/blob/0962a019148c102ca6f06c2d35e7f45aac62ca03/buildSrc/src/main/resources/gocd-docker-agent/docker-entrypoint.sh#L145

**Comment by rikostave1234** [comment]
@arvindsv Done :)

**Comment by arvindsv** [comment]
Looks good to me. :) Merged. I think new images will get built automatically, but the official released version will only be in 19.11.
